File: 1894-merge-strings-alternately/1894-merge-strings-alternately.ts
function mergeAlternately(word1: string, word2: string): string {
let merge: string = '';
let i:number = 0,j:number = 0;
while (i<word1.length || j<word2.length){
    if (i<word1.length){
        merge += word1[i++];
    }
    if (j<word2.length){
        merge += word2[j++];
    }
}
return merge;
};

File: 1894-merge-strings-alternately/1894-merge-strings-alternately.py
class Solution:
    def mergeAlternately(self, word1: str, word2: str) -> str:
        merge = []
        i,j = 0,0
        len1,len2 = len(word1),len(word2)

        while i<len1 or j<len2:
            if i<len1:
                merge.append(word1[i])
                i += 1
            if j<len2:
                merge.append(word2[j])
                j += 1
        return ''.join(merge)